VideoLocker/src/main/java/org/edx/mobile/module/db/impl/DbOperationBase.java

```diff
@@ -38,7 +38,7 @@ public T requestExecute(SQLiteDatabase db) {
             }
         }
         
-        return null;
+        return getDefaultValue();
```

Comment:
@hanningni: The issue here is not the population of entries with null values, but rather queries with null arguments, which cause an exception. I have change the exception handling logic to return a sane value, as there is no point in swallowing the exceptions if we are just to return null then. There is definitely no point for returning null where we expect `Integer`, `Long`, or `Boolean` values, and it was simply because of the lack of existing framework that this was implemented. For `String` values there can be a case for returning null, but an empty string is safer and I don’t see where it can cause any issues that a null value won’t cause.
> On 29-Jun-2015, at 7:00 pm, hanningni <notifications@github.com> wrote:
> 
> In VideoLocker/src/main/java/org/edx/mobile/module/db/impl/DbOperationBase.java <https://github.com/edx/edx-app-android/pull/280#discussion_r33466101>:
> 
> > @@ -38,7 +38,7 @@ public T requestExecute(SQLiteDatabase db) {
> >              }
> >          }
> >          
> > -        return null;
> > +        return getDefaultValue();
> this change may break existing logic.
> 
> because null is different from default value.
> 
> we should valid the data model before insert it into the database table.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/edx/edx-app-android/pull/280/files#r33466101>.
>